commons-test/src/main/java/org/infinispan/commons/test/TestNGNameTranslator.java

```diff
@@ -0,0 +1,80 @@
+package org.infinispan.commons.test;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import org.testng.ITestResult;
+import org.testng.annotations.Test;
+
+/**
+ * When a test is annotated with {@code @Test(testName = "foo")}, TestNG reports all the test methods in the class as
+ * having name "foo", ignoring the data provider parameters and the method name reported by {@code NamedTestMethod}
+ * (which includes the AbstractInfinispanTest parameters).
+ *
+ * Also, when running the same test but with a different module, only one test will be reported. For the JCache module,
+ * the same test will be executed in the remote and embedded module. In this case, we add the module name in the test
+ * name in order to find out what was the module that failed.
+ */
+public class TestNGNameTranslator {
+
+   public String translanteTestName(ITestResult res) {
+      StringBuilder result = new StringBuilder(res.getMethod().getMethodName());
+      if (res.getMethod().getConstructorOrMethod().getMethod().isAnnotationPresent(Test.class)) {
+         String dataProviderName = res.getMethod().getConstructorOrMethod().getMethod().getAnnotation(Test.class)
+               .dataProvider();
+         // Add parameters for methods that use a data provider only
+         if (res.getParameters().length != 0 && (dataProviderName != null && !dataProviderName.isEmpty())) {
+            result.append("(").append(deepToStringParameters(res));
+         }
+         // Add number of invocations to method name
+         if (res.getMethod().getConstructorOrMethod().getMethod().getAnnotation(Test.class).invocationCount() > 1) {
+            if (result.indexOf("(") == -1) {
+               result.append("(");
+            } else {
+               result.append(", ");
+            }
+            result.append("invoked ").append(
+                  res.getMethod().getConstructorOrMethod().getMethod().getAnnotation(Test.class).invocationCount())
+                  .append(" times");
+         }
+         // JCache tests are a special case
+         if (getModuleSuffix().contains("jcache")) {
```

Comment:
I haven't noticed before, but this step should be moved outside the parent if (`if (res.getMethod().getConstructorOrMethod().getMethod().isAnnotationPresent(Test.class))`) because it doesn't work on the `@Test` annotation.

Reply:
Good catch but now we have 3 new classes `JUnitNameTranslator`, `JUnitTestNameListener`, `ModuleNameTranslator` and 1 new property `inner.infinispan.module-suffix` because JCache TCK tests are JUnit based.

Now the console will print the right module under the test but the XML still has the name of the method. `JUnitTestNameListener` is not working because Surefire is responsible to generate the XML and is using the method name.

Reply:
`JUnitTestNameListener` was removed and `reportNameSuffix` was introduced. It will add the suffix at the end of the class name:

```xml
  <testcase name="testEqualsForAccessedExpiryPolicy" classname="javax.cache.expiry.ExpiryPolicyTest(embedded)" time="0.001"/>
  <testcase name="testModifiedExpiryPolicy" classname="javax.cache.expiry.ExpiryPolicyTest(embedded)" time="0.021"/>
  <testcase name="testEqualsForModifiedExpiryPolicy" classname="javax.cache.expiry.ExpiryPolicyTest(embedded)" time="0.018"/>
```

I believe that once we have https://issues.apache.org/jira/browse/SUREFIRE-1696 fixed we could remove the attribute `JUnitTestListener#progressLogger`